qiskit_dynamics/solvers/solver_functions.py

```diff
@@ -273,6 +282,17 @@ def rhs(t, y):
 
     if method == "scipy_expm":
         results = scipy_expm_solver(solver_generator, t_span, y0, t_eval=t_eval, **kwargs)
+    if method == "lanczos_diag":
+        if "sparse" not in generator.evaluation_mode:
```

Comment:
Note that `generator` here could be an arbitrary function, so you can't rely on it being one of the model classes. To raise this warning you'll need to first verify that `generator` is an instance of a `BaseGeneratorModel`. If it isn't you unfortunately won't be able to warn the user.

Reply:
fixed.